### ISSUE TITLE
universal linking not working on android 12

### ISSUE THREAD

**Issue by ChunChiehKung** [helium/maker-starter-app#36]
Hi Sir,

There is a 404 error when the Maker App try to use deep link to Helium App.
![78623](https://user-images.githubusercontent.com/29647730/164351511-9504c7bc-71df-4142-8c12-cccaaa4da346.jpg)

### We use the Android ADB to catch the log as below:(However, it looks not related to the 404 error)
transform[stderr]: https://github.com/browserslist/browserslist#browsers-data-updating
 BUNDLE  ./index.js 

 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `addListener` method.
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `removeListeners` method.
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `addListener` method.
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `removeListeners` method.
 LOG  Running "XXXXXX" with {"rootTag":1}
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `addListener` method. 
    at LanguageProvider (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.XXXXXX.makerapp&modulesOnly=false&runModule=true:313219:24)
    at render
    at RCTView
    at View
    at RCTView
    at View
    at AppContainer (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.XXXXXX.makerapp&modulesOnly=false&runModule=true:76768:36)
    at XXXXX(RootComponent) (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.XXXXXX.makerapp&modulesOnly=false&runModule=true:83149:28)
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `removeListeners` method. 
    at LanguageProvider (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.XXXXXX.makerapp&modulesOnly=false&runModule=true:313219:24)
    at render
    at RCTView
    at View
    at RCTView
    at View
    at AppContainer (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.XXXXXX.makerapp&modulesOnly=false&runModule=true:76768:36)
    at XXXXXX(RootComponent) (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.XXXXXX.makerapp&modulesOnly=false&runModule=true:83149:28)
 LOG  create ble manager
 LOG  setting ble log level to verbose
 LOG  manager create
 **_ERROR  Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function._**
    in HeliumActionSheet (at MoreListItem.tsx:99)

### The App Screenshot:
![unnamed (2)](https://user-images.githubusercontent.com/29647730/164352616-54095b51-5da8-4f32-a660-df1b9c645901.jpg)

**Comment by tyler-whitman** [comment]
We found the issue and it will be resolved soon, auto verification of our app links was failing. In the meantime users are able to manually verify the app link until the auto verification is working properly.

![enable_link](https://user-images.githubusercontent.com/6345962/164815540-97fff2fe-8b39-4113-a216-d72553e9e15a.gif)

**Comment by xiaoye-x** [comment]
I tried to manually set the link in Helium but the problem is the same. If I use sensecap: https://docs.sensecapmx.com/m1/sensecap-hotspot-app/sensecap-hotspot-app . It works ok, no need to manually set the link. So maybe there is some configure error in this Maker APP?

**Comment by tyler-whitman** [comment]
Yes some config must be incorrect in your maker app. I just verified on Android 12 and this is now resolved, the verified link is working properly on install of the Helium app. If you are still having issues please check the deeplink settings shown above and add helium.com as a link or reinstall the helium app which will check the verified links again and enable helium.com.

If you are still having the problem you must have something wrong with your deeplink config in your maker app.

**Comment by xiaoye-x** [comment]
Can you specify the detail of the deep link that need to use? The best is the current source code already has the correct one already.

**Comment by tyler-whitman** [comment]
> Can you specify the detail of the deep link that need to use? The best is the current source code already has the correct one already.

So the link doesn't open when using your version of the maker app, but it does when using sensecap? That is pretty odd since handling of the `helium.com` link should be the same if you have the Helium App installed. With a fresh clone of the maker app is it working? I just tested and it does for me. Do you have another phone you can try?

You need to change `makerappscheme` to whatever you want your maker app scheme to be. `makerappscheme` is the example and should not be used. If you search for `makerappscheme` you will see the integration of deep linking. You should also review https://developer.android.com/training/app-links/deep-linking and https://developer.android.com/training/app-links for Android help.
